src/RPCClient.ts

```diff
@@ -489,11 +489,12 @@ class RPCClient<M extends ClientManifest> {
         abortProm.p,
       ]);
       const tempWriter = rpcStream.writable.getWriter();
+      const id = this.idGen ? await this.idGen() : null;
       const header: JSONRPCRequestMessage = {
         jsonrpc: '2.0',
         method,
         params: headerParams,
-        id: null,
+        id,
```

Comment:
Don't do this.

Specify `idGen` as a parameter on creation and constructor.

Give it a default function of `() => null`.

Do the same for `RPCServer`.